api4/system.go

```diff
@@ -183,6 +183,17 @@ func getSystemPing(c *Context, w http.ResponseWriter, r *http.Request) {
 		w.Header().Set(filestoreStatusKey, s[filestoreStatusKey])
 	}
 
+	if deviceID := r.FormValue("device_id"); deviceID != "" {
+		canSend, err := c.App.SendTestPushNotification(deviceID)
+		if err != nil {
+			s["CanReceiveNotifications"] = err.Error()
```

Comment:
We don't do the same for other keys in the map. To remain consistent, I'd suggest to log a warning and return false instead.

Reply:
The problem is that returning false will mean "You cannot receive push notifications from this server". And that is a value that will be stored in the client "forever". This error may be something temporal. That will translate into the 3 states possible in the client: Can receive, cannot receive or unknown. We could hide the error from the client, by sending directly unknown, but not sure if we want to give more information to the user.

About logging or not a warning, I have no problem in doing that.

Reply:
> We could hide the error from the client, by sending directly unknown,

I like it, let's do that :+1: